### PR TITLE
Moving from Version to Build in ReferenceDocs

### PR DESCRIPTION
**Pull request by ldematte**
Generate the version component of the docs URLs using Build instead of Version.
Replicates the current logic, plus pointing to latest doc version ("current") for non-semantic versioning. 
Also, updates BaseRunAsSuperuserCommand to use RefereceDocs instead of its own logic.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/ReferenceDocs.java [hunk 1]
```diff
@@ -108,11 +110,31 @@ static Map<String, String> readLinksBySymbol(InputStream inputStream) throws Exc
...
+            if (isSnapshot && "0".equals(revision)) {
+                return UNRELEASED_VERSION_COMPONENT;
+            }
+            return major + "." + minor;
+        }
+        // Non-semantic version
+        if (isSnapshot) {
+            return UNRELEASED_VERSION_COMPONENT;
+        }
+        return CURRENT_VERSION_COMPONENT;
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/common/ReferenceDocs.java:137]: What is responsible for turning `current` into a sensible url?

Comment (ldematte) [reply server/src/main/java/org/elasticsearch/common/ReferenceDocs.java:137]: I don't think I understand your question fully. `current` points at the latest doc version, e.g https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-plugins.html 
I think is a good fallback for when we have no semantic version (e.g. serverless) - it makes sense to point to the latest version.

Comment (ldematte) [reply server/src/main/java/org/elasticsearch/common/ReferenceDocs.java:137]: Do you think it makes more sense to always point to `master` instead?

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/common/ReferenceDocs.java:137]: Ah right, `current` is valid in itself

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/common/ReferenceDocs.java:137]: Might be worth a comment indicating this is the expected scenario for hitting this, and why `current` is the correct place to go

**Review by thecoop — APPROVED** [review]

**Comment by ldematte** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-1

**Pull request merged**